pkg/storaged/fsys-is-empty.sh

```diff
@@ -0,0 +1,29 @@
+#! /bin/bash
+
+set -eux
+
+dev=$1
+
+need_unmount=""
+mp=$(findmnt -no TARGET "$dev" | cat)
+if [ -z "$mp" ]; then
+    mp=$(mktemp -d)
+    need_unmount=$mp
+    mount "$dev" "$mp" -o ro
+fi
+
+# A filesystem is empty if it only has directories in it.
+
+first=$(find "$mp" -not -type d | head -1)
+info=$(stat -f -c '{ "unit": %S, "free": %f, "total": %b }' "$mp")
```

Comment:
Neither of these commands is allowed to fail as then we will exit and never unmount if needed. So maybe we should use a trap for that.

(I can only get find to fail when not running as root for example but maybe it can fail on broken symlinks or pipes as root?)

Reply:
> So maybe we should use a trap for that.

Good point!